### PR TITLE
feat: add `ForgotPasswordSection` from Soul

### PR DESCRIPTION
**Pull request by apledger**
## What/Why?
Replace forgot password form with Soul components.

## Testing
Locally. Tested success and error messages.

Note: Captcha is missing and will be added in a later PR across all forms. See [linear task](https://linear.app/makeswift/issue/VIB-982/determine-strategy-for-recaptcha).

### CODE REVIEW

**Review by jorgemoya — APPROVED** [review]

File: core/app/[locale]/(default)/(auth)/login/forgot-password/_actions/reset-password.ts [hunk 1]
```diff
@@ -42,32 +24,29 @@ const ResetPasswordMutation = graphql(`
...
-  reCaptchaToken: string;
-}
-
-export const resetPassword = async ({
-  formData,
-  path,
-  reCaptchaToken,
-}: SubmitResetPasswordForm) => {
+export const resetPassword = async (
+  _lastResult: (SubmissionResult & { successMessage?: string }) | null,
```
Comment (migueloller) [inline core/app/[locale]/(default)/(auth)/login/forgot-password/_actions/reset-password.ts:28]: We should avoid merging into the Conform submission result and instead should wrap it like we did for other actions:

```ts
return { lastResult: submission.reply(), successMessage }
```

File: core/app/[locale]/(default)/(auth)/login/forgot-password/_components/reset-password-form/fragment.ts
Comment (migueloller) [inline core/app/[locale]/(default)/(auth)/login/forgot-password/_components/reset-password-form/fragment.ts:1]: We're going to want to keep this around/bring it back for captcha support

File: core/vibes/soul/form/form-status/index.tsx
Comment (migueloller) [inline core/vibes/soul/form/form-status/index.tsx:1]: Do we have a PR in VIBES with these changes?

**Review by migueloller — CHANGES_REQUESTED** [review]
Looking good! Only thing I think we need to change is payload of the action

File: core/vibes/soul/form/form-status/index.tsx
Comment (apledger) [reply core/vibes/soul/form/form-status/index.tsx:1]: Yes.

File: core/app/[locale]/(default)/(auth)/login/forgot-password/_actions/reset-password.ts (hunk 1 above)
Comment (apledger) [reply core/app/[locale]/(default)/(auth)/login/forgot-password/_actions/reset-password.ts]: Fixed.

File: core/app/[locale]/(default)/(auth)/login/forgot-password/_components/reset-password-form/fragment.ts
Comment (apledger) [reply core/app/[locale]/(default)/(auth)/login/forgot-password/_components/reset-password-form/fragment.ts:1]: No we don't. It's already commented out in the action.

Comment (apledger) [reply core/app/[locale]/(default)/(auth)/login/forgot-password/_components/reset-password-form/fragment.ts:1]: I've inlined the fragment into the query which is how it was in `vibes/soul-with-makeswift`

**Review by migueloller — APPROVED** [review]

**Pull request merged**